field_utils/FieldBundleDelta.F90

```diff
@@ -0,0 +1,306 @@
+
+      ! unique attribute in bundle
+      call update_interpolation_weights(this%interpolation_weights, bundle, ignore=ignore_, _RC)
+
+      _RETURN(_SUCCESS)
+   contains
+
+      subroutine update_units(units, field, ignore, rc)
+         character(*), optional, intent(in) :: units
+         type(ESMF_Field), intent(inout) :: field
+         character(*), intent(in) :: ignore
+         integer, optional, intent(out) :: rc
+
+         integer :: status
+         integer :: i
+         type(ESMF_Field), allocatable :: fieldList(:)
+
+         _RETURN_UNLESS(present(units))
+         _RETURN_IF(ignore == 'units')
+
+         call MAPL_FieldBundleGet(bundle, fieldList=fieldList, _RC)
+         do i = 1, size(fieldList)
+            call MAPL_InfoSetInternal(fieldList(i), key=KEY_UNITS, value=units, _RC)
+         end do
+
+         _RETURN(_SUCCESS)
+      end subroutine update_units
+
+      subroutine update_interpolation_weights(interpolation_weights, bundle, ignore, rc)
+         real(ESMF_KIND_R4), optional, intent(in) :: interpolation_weights(:)
+         type(ESMF_FieldBundle), intent(inout) :: bundle
+         character(*), intent(in) :: ignore
+         integer, optional, intent(out) :: rc
+
+         integer :: status
+
+         _RETURN_UNLESS(present(interpolation_weights))
+         _RETURN_IF(ignore == 'interpolation_weights')
+
+         call MAPL_InfoSetInternal(bundle, KEY_INTERPOLATION_WEIGHTS, values=interpolation_weights, _RC)
+
+         _RETURN(_SUCCESS)
+      end subroutine update_interpolation_weights
+
+   end subroutine update_bundle
+
+
+   ! If the size of the bundle is not changing, then any reallocation is
+   ! relegated to fields through the FieldDelta component.
+   ! Otherwise we need to create or destroy fields in the bundle.
+   
+   subroutine reallocate_bundle(this, bundle, ignore, unusable, rc)
+      class(FieldBundleDelta), intent(in) :: this
+      type(ESMF_FieldBundle), intent(inout) :: bundle
+      character(*), intent(in) :: ignore
+      class(KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      type(ESMF_Field), allocatable :: fieldList(:)
+      type(ESMF_Geom) :: bundle_geom
+      integer :: i
+      type(LU_Bound), allocatable :: bounds(:)
+      type(LU_Bound) :: vertical_bounds
+      type(ESMF_TypeKind_Flag) :: typekind
+      integer, allocatable :: ungriddedLbound(:), ungriddedUbound(:)
+      type(ESMF_Info) :: ungridded_info
+      type(ESMF_Info) :: vertical_info
+      integer :: old_field_count, new_field_count
+      integer :: num_levels
+      character(:), allocatable :: units, vloc
+      character(ESMF_MAXSTR), allocatable :: fieldNameList(:)
+
+      ! Easy case 1: field count unchanged
+      call MAPL_FieldBundleGet(bundle, fieldList=fieldList, _RC)
+      _RETURN_UNLESS(allocated(this%interpolation_weights))
+      new_field_count = size(this%interpolation_weights) - 1
```

Comment:
Why is the number of new fields one less than the number of interpolation_weights?

Reply:
The first element of the weights is a constant offset.   This is to handle the case when the rule is to set an ext data export to a constant value.   Easiest way to handle this was just to represent with empty bundle.  I will add a comment in source.